src/averageAges.js

```diff
@@ -37,7 +41,15 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const onlyWomans = people.filter(({ sex }) => sex === 'f');
+  const womanWithChildren = onlyWomans.filter(woman =>
+    people.some(child => child.mother === woman.name));
+  const currentSearch = withChildren ? womanWithChildren : onlyWomans;
+
+  const womansAge = currentSearch.map(({ born, died }) => died - born);
+  const sumOfAges = womansAge.reduce((sum, current) => sum + current);
```

Comment:
You do the same on `calculateMenAverageAge`. You can create a separate function and use it here and in `calculateMenAverageAge`.